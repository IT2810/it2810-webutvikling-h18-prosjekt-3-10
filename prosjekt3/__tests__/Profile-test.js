import React from 'react';
import renderer from 'react-test-renderer';
import ProfileScreen from '../screens/ProfilesScreen';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import { AsyncStorage } from 'react-native';
import MockAsyncStorage from 'mock-async-storage';



const mock = () => {
    const mockImpl = new MockAsyncStorage()
    jest.mock('AsyncStorage', () => mockImpl)
}
const release = () => jest.unmock('AsyncStorage')

describe('ProfileScreen snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders the profile screen', async () => {
        const tree = renderer.create(<ProfileScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});


describe('Unit testing', () => {
    let profileScreen = renderer.create(<ProfileScreen />).getInstance();
    const user = {
        name: 'Jens',
        email: 'test@testorini.com',
        town: 'Trondheim',
        myHeightNumber: '180',
        myWeightNumber: '80',
        modalVisible: false,
    }

    it('should save profileuser in state', async () => {
        mock();
        await profileScreen.saveState(user);
        const value = await AsyncStorage.getItem('profile')
        //expect(profileScreen.state.saveState.length).toEqual(1);
        expect(value).toBe(JSON.stringify(profileScreen))
    })


    release();

});

