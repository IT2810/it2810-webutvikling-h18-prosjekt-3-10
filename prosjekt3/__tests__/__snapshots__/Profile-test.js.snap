// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen snapshot renders the profile screen 1`] = `null`;

exports[`renders button with passed props 1`] = `
<View
  enabled={true}
  keyboardVerticalOffset={0}
  onLayout={[Function]}
>
  <View
    style={
      Array [
        undefined,
        Object {
          "bottom": 0,
        },
      ]
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  enabled={true}
  keyboardVerticalOffset={0}
  onLayout={[Function]}
>
  <View
    style={
      Array [
        undefined,
        Object {
          "bottom": 0,
        },
      ]
    }
  />
</View>
`;
